common/main/java/com/couchbase/lite/AbstractQuery.java

```diff
@@ -137,7 +137,7 @@ public void setParameters(@Nullable Parameters parameters) throws CouchbaseLiteE
 
             if (parameters == null) { return; }
 
-            try { getC4QueryLocked().setParameters(parameters.encode()); }
+            try (FLSliceResult encodedParams = parameters.encode()) { getC4QueryLocked().setParameters(encodedParams); }
```

Comment:
From the code, I couldn't tell whether setParameters will copy bytes or retain the FLSliceResult or not. Can you confirm if it is doing one of the above?

Reply:
`setParameters` is a call to `c4query_setParameters`, which takes as an argument, a `C4String`.  According to @snej , there is no need to worry about C4String parameters, after a call returns.